src/clj/util/time.clj

```diff
@@ -8,3 +8,22 @@
    :minutes."
   [{:keys [year month day hour minute]}]
   (java.util.Date. (- year 1900) month day hour minute))
+
+
+(defn is-date-object?
+  "Checks if provided argument
+   is an instance of the java.util.Date
+   class."
+  [timestamp]
+  (instance? java.util.Date timestamp))
+
+
+(defn start-before-end?
+  "Returns true if bot arguments
+   are java.util.Date instances
+   and the start date is earlier
+   than the end date."
+  [start end]
+  (and (is-date-object? start)
+       (is-date-object? end)
+       (= -1 (compare start end))))
```

Comment:
Wäre hier nicht theoretisch besser < 0 zu checken anstatt genau auf -1? Weil bei einem drei Wege vergleich kann das Ergebnis ja auch einfach nur negativ sein.

Reply:
compare kann nur -1, 0 und 1 zurückgeben, wobei -1 der Fall ist wenn Start früher das Ende ist.
Deswegen wäre < 0 und = -1 in dem Fall äquivalent.
Oder habe ich die Frage anders verstanden als du sie meintest? :thinking:

Reply:
Kannst du mir eine Referenz dafür schicken? Weil in der [Doku von compare](https://clojuredocs.org/clojure.core/compare) steht folgendes:
> Comparator. Returns a negative number, zero, or a positive number
when x is logically 'less than', 'equal to', or 'greater than'
y. Same as Java x.compareTo(y) except it also works for nil, and
compares numbers and collections in a type-independent manner. x
must implement Comparable

Wenn ich nicht gerade auf die Falsche Funktion gucke, sollte man besser auf kleiner 0 checken als direkt auf -1.

Reply:
Ah ja, stimmt ich glaube das war ein sonderfall, dass bei Java.util.date instanzen nur die drei werte geworfen werden koennen. Ich aender es mal eben ab, danke :smile:

Reply:
Es wird jetzt nach < 0 geprueft :+1: